Wikipedia/Code/NSUserDefaults+WMFExtensions.swift

```diff
@@ -44,6 +44,8 @@ let WMFSearchLanguageKey = "WMFSearchLanguageKey"
 @objc public extension UserDefaults {
     @objc(WMFUserDefaultsKey) public class Key: NSObject {
         @objc static let defaultTabType = "WMFDefaultTabTypeKey"
+        @objc static let didShowLoggedOutPanel = "WMFDidShowLoggedOutPanelKey"
```

Comment:
Could the persistence be handled by one property on `WMFAuthenticationManager`? The underlying storage could still be user defaults, but would be good to have `WMFAuthenticationManager` own the state instead of having the panel check it. When the user is logged out by `.app` or `.server`, the property `isUserUnintentionallyLoggedOut` is set to `true`. `WMFAppViewController` checks this when it gets the logged out notification and also on launch after login is attempted. When the user acknowledges the alert, `WMFAppViewController` calls `userDidAcknowledgeUnintentionalLogout()` on `WMFAuthenticationManager` which resets `isUserUnintentionallyLoggedOut` to false.